lightning/cogs/automod/ui.py

```diff
@@ -191,3 +203,338 @@ async def warn_ban(self, itx: discord.Interaction, button: discord.ui.Button):
         self.choice = "warn_ban"
         await itx.response.edit_message(view=None)
         self.stop()
+
+
+class FakeCtx:
+    def __init__(self, member) -> None:
+        self.author = member
+
+
+class GatekeeperRoleView(BasicMenuLikeView):
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (testing):** Add tests for `GatekeeperRoleView` to verify role selection and permission checks.

Given the critical role of `GatekeeperRoleView` in setting up gatekeeper roles, it's important to verify through tests that the role selection works correctly and that the permission checks prevent unauthorized role assignments.